src/globus_sdk/services/timer/data.py

```diff
@@ -0,0 +1,133 @@
+import datetime
+from typing import Any, Dict, Optional, Union
+
+from globus_sdk.config import get_service_url
+from globus_sdk.services.transfer import TransferData
+from globus_sdk.utils import PayloadWrapper, slash_join
+
+
+class TimerJob(PayloadWrapper):
+    r"""
+    Class for specifying parameters used to create a job in the Timer service. Used as
+    the ``data`` argument in
+    :meth:`create_job <globus_sdk.TimerClient.create_job>`.
+
+    Timer operates through the `Globus Automate API
+    <https://docs.globus.org/globus-automation-services/>`_. Crucially, the
+    ``callback_url`` parameter should always be the URL used to run an action provider.
+
+    .. warning::
+
+        Currently the only supported action provider for this is for Transfer. Thus,
+        users should generally only use the :meth:`~from_transfer_data` method here. Any
+        other usage is meant for internal purposes; proceed with caution!
+
+    :param callback_url: URL for the action which the Timer job will use.
+    :type callback_url: str
+    :param callback_body: JSON data which Timer will send to the Action Provider on
+        each invocation
+    :type callback_body: dict
+    :param start: The datetime at which to start the Timer job.
+    :type start: datetime.datetime or str
+    :param interval: The interval at which the Timer job should recur.
+    :type interval: datetime.timedelta
```

Comment:
I just noticed this, but should this be documented as `timedelta or int` and "The interval in seconds ..."?
Otherwise, the only way I'd know what `interval=300` means is by reading source here and seeing that the delta is converted to seconds.

Also applies to the same param in the Transfer helper.